libretroshare/src/pgp/pgphandler.cc

```diff
@@ -1381,10 +1381,12 @@ bool PGPHandler::decryptTextFromFile(const RsPgpId&,std::string& text,const std:
 
 bool PGPHandler::SignDataBin(const RsPgpId& id,const void *data, const uint32_t len, unsigned char *sign, unsigned int *signlen,bool use_raw_signature, std::string reason /* = "" */)
 {
-	RsStackMutex mtx(pgphandlerMtx) ;				// lock access to PGP memory structures.
```

Comment:
Moving this mutex is really not a good idea: locked_getSecretKey(id) returns a pointer in the middle of a large array containing all key structures. If anything happens to the keyring between the getSecretKey and the use of it, then an undefined behavior will happen.